File: palindromic-substrings/palindromic-substrings.py
class Solution(object):
    def countSubstrings(self, s):
        ans=0
        for i in range(len(s)):
            for j in range(i,len(s)):
                temp=s[i:j+1]
                #print(temp)
                ans+=(temp==temp[::-1])
        return ans

File: palindromic-substrings/palindromic-substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        
        int n=s.size();
        int ans=0;
        vector <vector <int>> dp(n,vector <int> (n,0));
        for(int i=n-1;i>=0;i--)
        {
          for(int j=i;j<n;j++)
          {
              dp[i][j]=s[i]==s[j]&&((j-i+1<=2)||dp[i+1][j-1]);
              ans+=dp[i][j];
          }
        }
        
        return ans;
    }
};